compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaFunctionTypeSymbol.java

```diff
@@ -133,7 +133,12 @@ public BallerinaFunctionTypeSymbol(CompilerContext context, ModuleID moduleID,
     public Optional<TypeSymbol> returnTypeDescriptor() {
         if (returnType == null) {
             TypesFactory typesFactory = TypesFactory.getInstance(this.context);
-            this.returnType = typesFactory.getTypeDescriptor(this.typeSymbol.returnType);
+            if (this.getBType().getReturnType() != null) {
+                this.returnType = typesFactory.getTypeDescriptor(this.getBType().getReturnType());
+            } else {
+                this.returnType = typesFactory.getTypeDescriptor(this.typeSymbol.returnType);
```

Comment:
We can improve this one by extracting `this.getBType().getReturnType()` to a variable. (minor)

Reply:
Fixed.